jwst/outlier_detection/outlier_detection_tso.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+from .outlier_detection import OutlierDetection, flag_cr
+from jwst.resample.resample_utils import build_mask
+
+from jwst import datamodels as dm
+import logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+__all__ = ["OutlierDetectionTSO"]
+
+
+class OutlierDetectionTSO(OutlierDetection):
+    """Class to flag outlier pixels in DQ of TSO data. Works similarly to
+    imaging outlier detection, but does not resample and uses a rolling median."""
+
+    def __init__(self, input_model: dm.CubeModel | dm.ModelContainer, **pars):
+        """Initialize class for TSO data processing.
+
+        Parameters
+        ----------
+        input_models : ~jwst.datamodels.CubeModel or ~jwst.datamodels.ModelContainer
+            The input TSO data cube. if a ModelContainer is passed in, it is assumed
+            to be a list of ImageModels, and gets converted to a CubeModel.
+
+        """
+        super().__init__(input_model, **pars)
+        if isinstance(self.inputs, dm.ModelContainer):
+            self._undo_convert_inputs()
```

Comment:
I put this in because the unit tests in test_outlier_detection.py were failing.  Those set up mock data to pass into the step that are in a ModelContainer of ImageModels rather than in a CubeModel.

Of course maintaining consistency with unit tests is not a good reason to keep bad code around in itself, but the test highlights that this is an API change: it used to be possible to pass a ModelContainer directly into the outlier detection step, and now it isn't.

Thinking about this more, and given that build 11.0 is incremented as a "major release", I agree with you that allowing a ModelContainer to pass directly into this step is probably not necessary and shouldn't be supported.